mlir/test/Catalyst/ConversionTest.mlir

```diff
@@ -128,3 +128,34 @@ module @test0 {
   catalyst.callback @callback_4(memref<f64>, memref<f64>) attributes {argc = 2 : i64, id = 4 : i64, resc = 3 : i64}
 }
 
+// -----
+
+// CHECK-LABEL: @test1
+module @test1 {
+  catalyst.callback @callback_1(memref<f64>, memref<f64>) attributes {argc = 1 : i64, id = 1 : i64, resc = 1 : i64}
+  // CHECK-LABEL: func.func private @foo(
+  // CHECK-SAME: [[arg0:%.+]]: tensor<f64>
+  // CHECK-SAME:)
+  func.func private @foo(%arg0: tensor<f64>) -> tensor<f64> {
+    // CHECK: [[memref0:%.+]] = bufferization.to_memref [[arg0]]
+    %0 = bufferization.to_memref %arg0 : memref<f64>
+    // CHECK: [[struct0:%.+]] = builtin.unrealized_conversion_cast [[memref0]]
+    // CHECK: [[tensor1:%.+]] = bufferization.alloc_tensor()
+    %1 = bufferization.alloc_tensor() {memory_space = 0 : i64} : tensor<f64>
+    // CHECK: [[memref1:%.+]] = bufferization.to_memref [[tensor1]]
+    %2 = bufferization.to_memref %1 : memref<f64>
+    // CHECK: [[struct1:%.+]] = builtin.unrealized_conversion_cast [[memref1]]
+
+    // CHECK: [[ptr0:%.+]] = llvm.alloca {{.*}}
+    // CHECK: llvm.store [[struct0]], [[ptr0]]
+    // CHECK: [[ptr1:%.+]] = llvm.alloca {{.*}}
+    // CHECK: llvm.store [[struct1]], [[ptr1]]
```

Comment:
This can be optimized. No need to allocate memory on the stack, just to store the struct0 which will come from a ptr.